app/controllers/onboardings_controller.rb

```diff
@@ -20,6 +20,16 @@ def continue
 
   def select_plan
     change_current_step(2)
+    @account = onboarding.account
+    payment_recurrence = PaymentRecurrence.includes(:plans)
+    @account_plan = @account.plan || @account.build_plan(payment_recurrence: payment_recurrence.last)
+
+    @monthly_plans = payment_recurrence.find_by(acronym: 'mo').plans.where(status: 'available')
+    @yearly_plans = payment_recurrence.find_by(acronym: 'yo').plans.where(status: 'available')
+  end
```

Comment:
You could extract these model queries into scopes and use them here in the controller. That would make the controller easier to read and separate concerns in a better way
```ruby
scope :with_monthly_plans, -> { includes(:plans).find_by(acronym: 'mo').plans.where(status: 'available') }
scope :with_yearly_plans, -> { includes(:plans).find_by(acronym: 'yo').plans.where(status: 'available') }
```

```suggestion
  def select_plan
    change_current_step(2)
    @account = onboarding.account
    @account_plan = @account.plan || @account.build_plan(payment_recurrence: PaymentRecurrence.last)

    @monthly_plans = PaymentRecurrence.with_monthly_plans
    @yearly_plans = PaymentRecurrence.with_yearly_plans
  end
```